### DISCUSSION TITLE
USB audio support

### DISCUSSION THREAD

**Discussion by unmanagedtn** [devmfc/debian-on-amlogic#64] [Ideas]
Any chance the basic kernal options be added?

See https://www.kernelconfig.io/config_snd_usb

Working on a project to take a rack mount weather radio and use spchcat to convert the audio to text then send it to an digital tv encoder for my own in house weather station onthe tv... And then send text based alerts to my cellphone also, but again need mic input.

ALSA can be compiled with the generic snd-usb-audio module and it looks like the soundcore module ia already there....

I have acquired about 12 of these set boxes. And thanks again for all your hard work.

**Comment by devmfc** [comment]
Not really.... I try to only create a kernel with a minimum of drivers. Only the drivers needed for the Amlogic based TVBoxes.

But you can compile the needed modules yourself? The kernel headers are available for every release, so it shouldn't be that hard?

**Comment by unmanagedtn** [comment]
I can compile the modules myself if the kernel headers that are needed are
in the release as no extra work would be required by you and again thanks
for the hard work and being so responsive to my mundane questions.

**Comment by unmanagedtn** [comment]
Well I got the kernel and did a recompile of the moudles but it said exec format was wrong ... so I decied to update the kernel to fix and of course I bricked it ... now its stuck on the box logo in a loop of reboots

**Reply by devmfc** [reply]
Uhm.... let me guess, you didn't cross compile for arm64? 

Anyway, just burn a fresh image on USB and boot from that, then you can repair your system on eMMC.

**Comment by unmanagedtn** [comment]
Nope 👎 did not cross compile, going to start from scratch and learn by
force

**Comment by unmanagedtn** [comment]
I did the recompile of the kernel modules on the device would I still have
to cross-compile for arm64 even though it's an arm based device I'm new to
the arm world so again I apologize for such mundane idiotic questions.

**Comment by unmanagedtn** [comment]
Think I figured it out need to run

make modules_prepare

before compile....

Will test after work.

**Comment by unmanagedtn** [comment]
ok there are a few things diffrent ... I had to upgrade to your current release as there is a known bug with the older kernel soruce and it wold not compile but I got it to compile the sound modules.

**Comment by unmanagedtn** [comment]
I will make a how-to add upload it to my github account for others.

**Reply by heiko442** [reply]
Do you already added a How-To? Or could you just describe in a few lines what to do in order to get the sound modules? I'm struggling for some days now with the snd_usb_audio module and are not able to compile it.

**Reply by unmanagedtn** [reply]
A little more detail in where you are and are not getting this to work would help. Be warned I am starting a new project at work and wont be online much for the month.

**Reply by unmanagedtn** [reply]
Need to get headers from the releasaes and the sorurce for the kernel you are using and then run make menucofig and turn on usb soundcard support, you most copy over the config-X.X.X over to linux source directory so the settings for kernal config can be imported from menuconfig, and then after you save them before you exit, then you can use the "make" target "modules_prepare." This will make sure the kernel contains the information required. The target exists solely as a simple way to prepare a kernel source tree for building external modules. The in the soruce type make modules....  this all asumes you have installed all the packages needed to build a kernel...
I will try to get this up this weekend and I will need to document what I have found out for you. sorry for the delay life gets in the way of fun.... :)

**Comment by petesimon** [comment]
thank you all for adding usb audio to the latest images, in particular "Devmfc_Ubuntu-Oracular_6.6.60-meson64_Minimal-24.11.10.img.xz". i plugged in a generic audio dongle and it was immediately recognized as C-Media Electronics Inc USB PnP Sound Device 08bb:2902. i tested it by running `aplay /usr/share/sounds/alsa/Front_Center.wav` and it was too loud, hehe. so `alsamixer` works to lower the volume. i guess the mic input also works but didn't test it. the dongle was $2 USD plus tax from aliexpress and it has a red light that blinks when audio is being decoded. very nice. thanks again.
